cmd/enterprise-portal/internal/dotcomdb/dotcomdb_test.go

```diff
@@ -0,0 +1,277 @@
+package dotcomdb_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	pgxstdlibv4 "github.com/jackc/pgx/v4/stdlib"
+	"github.com/jackc/pgx/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/sourcegraph/log/logtest"
+
+	"github.com/sourcegraph/sourcegraph/cmd/enterprise-portal/internal/dotcomdb"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/dotcomproductsubscriptiontest"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/graphqlbackend"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbconn"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbtest"
+	"github.com/sourcegraph/sourcegraph/internal/license"
+	"github.com/sourcegraph/sourcegraph/internal/licensing"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+)
+
+func newTestDotcomReader(t *testing.T) (database.DB, *dotcomdb.Reader) {
+	ctx := context.Background()
+
+	// Set up a Sourcegraph test database.
+	sgtestdb := dbtest.NewDB(t)
+
+	// HACK: Extract the underlying pgx connection from the Sourcegraph test
+	// database, so that we can tease out the connection string that was used
+	// to connect to it, since we want to use pgx/v5 directly as offered by MSP
+	// utilities. This is sneaky but easier than changing the extensive
+	// infrastructure around dbtest
```

Comment:
Super minor nitpick: Long form comments like this should end with terminal punctuation.

```suggestion
	// infrastructure around dbtest.
```

Reply:
Here and throughout the file. 👉 👈 Please.